src/VoteProxy2.sol

```diff
@@ -0,0 +1,48 @@
+pragma solidity ^0.4.24;
+
+import 'ds-token/token.sol';
+import 'ds-chief/chief.sol';
+
+contract VoteProxy2 {
+    address public cold;
+    address public hot;
+    DSToken public GOV;
+    DSToken public IOU;
+    DSChief chief;
+
+    constructor(DSChief chief_, address cold_, address hot_) {
+        chief = chief_;
+        cold = cold_;
+        hot = hot_;
+        GOV = chief.GOV();
+        IOU = chief.IOU();
+        GOV.approve(chief, uint256(-1));
+        IOU.approve(chief, uint256(-1));
+    }
+
+    modifier onlyCold() { require(msg.sender == cold); _ }
+    modifier hotOrCold() { require(msg.sender == cold || msg.sender == hot); _ }
+
+    function vote(bytes32 slate)
+        hotOrCold
+    {
+        chief.vote(slate);
+    }
+    function lock(uint256 wad)
+        hotOrCold
+    {
+        chief.lock(wad);
+    }
+    function free(uint256 wad)
```

Comment:
What is the reasoning for `lock` and `free` taking in a `wad`?  In what scenario would the `VoteProxy2` contract hold MKR but you would not want to lock or free it all?

I can _kind of_ see an argument for partial free, but it feels like `lock` should really be `gov.balanceOf(this)` instead of `wad` since I can't think of a reason why you would not want to lock all of it.  The argument for partial free is _weak_, but not completely unreasonable.

Reply:
I'm imagining an institutional user which has separate holding/voting controls, and still wants to demonstrate control of funds in their cold key. But also this solution seems to be simplest and also most general. Even a contract caller could do `proxy.free(gov.balanceOf(proxy));` if they want the whole amount.

Reply:
Actually, since we're talking about just lock/free and not release, my reasoning makes a bit less sense, but I still think we should keep it as is. It's simpler and more general, and doesn't block any use cases.

Reply:
Yeah, it certainly is _more_ permissive to support taking in a `wad`.  Just makes the API a bit more complicated to use is all.  Certainly not insurmountable, but depending on how you are calling this it may be an extra round-trip to a remote Ethereum node 🤷‍♂️

Reply:
Sticking with wads for now, can bring it up later. Maybe some auxiliary functions where we can drop args / compose primitive functions (e.g. old releaseAll)